flixel/input/gamepad/id/XInputID.hx

```diff
@@ -4,52 +4,210 @@ import flixel.input.gamepad.FlxGamepad;
-	public static var RIGHT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(2, 3);
-	
-	public static inline var LEFT_TRIGGER:Int = 4;
-	public static inline var RIGHT_TRIGGER:Int = 5;
-	
-	#if FLX_JOYSTICK_API
-	//the axis index values for this don't overlap with anything so we can just return the original values!
-	public static function axisIndexToRawID(index:Int):Int
-	{
-		return index;
-	}
+	#if flash
+		
+		// Button IDs
+		public static inline var A:Int = 4;
+		public static inline var B:Int = 5;
+		public static inline var X:Int = 6;
+		public static inline var Y:Int = 7;
+		public static inline var LB:Int = 8;
+		public static inline var RB:Int = 9;
+		
+		public static inline var BACK:Int = 12;
+		public static inline var GUIDE:Int = -1;
+		public static inline var START:Int = 13;
+		
+		public static inline var LEFT_STICK_CLICK:Int = 14;
+		public static inline var RIGHT_STICK_CLICK:Int = 15;
+		
+		public static inline var DPAD_UP:Int = 16;
+		public static inline var DPAD_DOWN:Int = 17;
+		public static inline var DPAD_LEFT:Int = 18;
+		public static inline var DPAD_RIGHT:Int = 19;
+		
+		// Axis indices
+		public static var LEFT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(0, 1);
+		public static var RIGHT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(2, 3);
+		
+		public static inline var LEFT_TRIGGER:Int = 10;
+		public static inline var RIGHT_TRIGGER:Int = 11;
+		
+		public static inline var LEFT_ANALOG_STICK_H = 0;
+		public static inline var LEFT_ANALOG_STICK_V = 1;
+		public static inline var RIGHT_ANALOG_STICK_H = 2;
+		public static inline var RIGHT_ANALOG_STICK_V = 3;
+		
+	#elseif FLX_GAMEINPUT_API
+		
+		// Button IDs
+		public static inline var A:Int = 6;
+		public static inline var B:Int = 7;
+		public static inline var X:Int = 8;
+		public static inline var Y:Int = 9;
+	
+		public static inline var BACK:Int = 10;
+		#if mac
+		public static inline var GUIDE:Int = 11;
+		#else
+		public static inline var GUIDE:Int = -1;
+		#end
+		public static inline var START:Int = 12;
+	
+		public static inline var LEFT_STICK_CLICK:Int = 13;
+		public static inline var RIGHT_STICK_CLICK:Int = 14;
+	
+		public static inline var LB:Int = 15;
+		public static inline var RB:Int = 16;
 	
-	//Just pass back LEFT/RIGHT trigger
-	public static inline var LEFT_TRIGGER_FAKE:Int = LEFT_TRIGGER;
-	public static inline var RIGHT_TRIGGER_FAKE:Int = RIGHT_TRIGGER;
+		public static inline var DPAD_UP:Int = 17;
+		public static inline var DPAD_DOWN:Int = 18;
+		public static inline var DPAD_LEFT:Int = 19;
+		public static inline var DPAD_RIGHT:Int = 20;
+	
+		// Axis indices
+		public static var LEFT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(0, 1);
+		public static var RIGHT_ANALOG_STICK(default, null) = new FlxGamepadAnalogStick(2, 3);
+	
+		public static inline var LEFT_TRIGGER:Int = 4;
+		public static inline var RIGHT_TRIGGER:Int = 5;
+	
+		
+	#elseif FLX_JOYSTICK_API
+		#if !web
```

Comment:
The way we currently have it set up, FLX_JOYSTICK_API basically just means "legacy," and there is no legacy target for "web" -- flash has no next/legacy distinction (or joystick API for that matter, it only has access to the gamepad API), and html5 *only* works with next (and I haven't checked but it's a solid bet that it only has access to the gamepad API as well).

The !web may very well be redundant, but it doesn't represent a hole -- there is no "legacy / joystick" + "flash / html5" mode. Flash target will take the #if flash path and HTML5 target will take the FLX_GAMEINPUT_API path.